include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1054,15 +1065,28 @@ namespace aux {
 				, error_code& ec
 				, int flags);
 
+			void send_udp_packet_hostname_listen(aux::session_listen_socket* sock
```

Comment:
I'm honestly not sure what is and isn't possible when it comes to proxy configurations. I just did this for constancy with the regular udp send function and because unless `m_force_proxy` is set we can still end up sending directly.